metron-platform/metron-elasticsearch/src/main/java/org/apache/metron/elasticsearch/writer/ElasticsearchWriterConfig.java

```diff
@@ -0,0 +1,134 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package org.apache.metron.elasticsearch.writer;
+
+import org.apache.metron.common.Constants;
+import org.apache.metron.stellar.common.utils.ConversionUtils;
+
+import java.util.Map;
+
+/**
+ * Configuration settings that customize the behavior of the {@link ElasticsearchWriter}.
+ */
+public enum ElasticsearchWriterConfig {
+
+  /**
+   * Defines which message field, the document identifier is set to.
+   *
+   * <p>If defined, the value of the specified message field is set as the Elasticsearch doc ID. If
+   * this field is undefined or blank, then the document identifier is not set.
+   */
+  DOC_ID_SOURCE_FIELD("es.document.id", "", String.class);
```

Comment:
I put this value here because I didn't see any other place for it at the time.  I then later found the other properties like `es.ip` defined in `ElasticsearchUtils`.  

Many of the property definitions there are hard coded and could use a clean-up. I'd like to do a follow-on to add `es.ip`, `es.date.format`, `es.client.class`, etc to this configuration class so that each is clearly defined.

As part of this PR, we could keep this property here or I could move it to `ElasticsearchUtils` to match those other properties.  Open to reviewer feedback on this.

Reply:
This does leave things in an inconsistent state and will makes things confusing for anyone working with these classes.  However I think a follow up PR would be fine.

Are you aware of the [ConfigOption](https://github.com/apache/metron/blob/master/metron-platform/metron-common/src/main/java/org/apache/metron/common/configuration/ConfigOption.java) interface?      It looks like there may be some duplication with this class.

Reply:
I didn't know we had `ConfigOption`.  I will remove `ElasticsearchWriterConfig` for now.  Then as a follow-on, I will do something similar, but reuse `ConfigOption`.  Thanks for the pointer!

Reply:
The last commit removes `ElasticsearchWriterConfig`.